cpp/dolfinx/mesh/Mesh.cpp

```diff
@@ -112,8 +117,19 @@ Mesh mesh::create(MPI_Comm comm,
     }
   }
 
+  // FIXME: improve. Clip cell data, removing any unused ghost cells
+  // before sending to create_geometry. Maybe AdjacencyList can have a "resize"
+  // method?
+  int n_cells_local = topology.index_map(tdim)->size_local()
+                      + topology.index_map(tdim)->num_ghosts();
+  Eigen::Matrix<std::int32_t, Eigen::Dynamic, 1> off1
+      = cell_nodes.offsets().head(n_cells_local + 1);
+  Eigen::Matrix<std::int64_t, Eigen::Dynamic, 1> data1
+      = cell_nodes.array().head(off1[n_cells_local]);
+  graph::AdjacencyList<std::int64_t> cell_nodes_2(data1, off1);
+
   Geometry geometry
-      = mesh::create_geometry(comm, topology, element, cell_nodes, x);
+      = mesh::create_geometry(comm, topology, element, cell_nodes_2, x);
```

Comment:
Or adapt this function such that it ignores ghost cells. Or maybe ghost cells are helpful?